src/libs/catalyst/AscentCatalyst.cxx

```diff
@@ -0,0 +1,178 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+
+#include <ascent.h>
+
+#include <catalyst.h>
+#include <catalyst_conduit.hpp>
+#include <catalyst_conduit_blueprint.hpp>
+#include <catalyst_stub.h>
+
+#include <conduit.hpp>
+#include <conduit_cpp_to_c.hpp>
+
+#include "catalyst_impl_ascent.h"
+
+#ifdef _USE_MPI
+#include <mpi.h>
+#endif
+
+namespace detail{
+
+class Instance
+{
+protected:
+    Instance() = default;
+
+private:
+    //pointer to the ascent instance
+    void* ascent = nullptr;
+    std::vector<conduit::Node> actions;
+    //instance pointer;
+    static Instance* instance;
+
+public:
+    // deleting copy constructor
+    Instance(const Instance& obj) = delete;
+
+    static Instance* GetInstance()
+    {
+      if(instance == nullptr)
+      {
+        instance = new Instance();
+        return instance;
+      }
+      else
+      {
+        return instance;
+      }
+    }
+
+    void SetAscent(void* _ascent) { this->ascent = _ascent; }
+    void* GetAscent() { return this->ascent; }
+
+    void AddActions(conduit::Node& node)
+    {
+      actions.push_back(node);
```

Comment:
```suggestion
      this->actions.emplace_back(node);
```

Reply:
Thanks for the suggestion; I've updated the code